### PR TITLE
Force env extension to the end, before user

### PR DESCRIPTION
**Pull request by galou**
Signed-off-by: Gaël Écorchard <gael.ecorchard@cvut.cz>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -31,15 +31,15 @@
 
 kwargs = {
     'name': 'rocker',
-    'version': '0.2.3',
+    'version': '0.2.4',
```
Comment (tfoote) [inline setup.py:34]: Same comments here about changing the version and the unrelated changes.

**Review by tfoote — CHANGES_REQUESTED** [review]
Quick cleanup on the patch, but otherwise this looks good. Moving the logic into the core makes sense. It would be great to add a quick unit test for this logic too to make sure it doesn't break with any future refactoring.

File: setup.py (hunk 1 above)
Comment (galou) [reply setup.py:34]: Sorry, I don't get it, the other version change was for the minor number but here I changed the patch version. Do you mean that I should let the version to 0.2.3?

**Review by galou — COMMENTED** [review]
I have to admit that I don't know how to write a unit test to test that the env extension is one before the last when it's supposed to be the sole extension tested.

**Comment by tfoote** [comment]
>  I should let the version to 0.2.3?

Correct. Your patch may be bundled in with other patches and then a release made. The version number will be bumped during the release process and is separate from your changes.

**Comment by galou** [comment]
Reverted the version number and unrelated changes. Should I squeeze commits?

**Comment by tfoote** [comment]
Thanks for the cleanup. 

> I have to admit that I don't know how to write a unit test to test that the env extension is one before the last when it's supposed to be the sole extension tested.

To write the test, you'll need to invoke it with a couple of extensions enabled and check the ordering. 

extending this test https://github.com/osrf/rocker/blob/280b5e13d3a6b85a872c2250dceb489b9ea06dcc/test/test_core.py#L123

Coverage should be decent with the following tests:
* push user to last
* push env to last except user
* push env to second to last with user
